restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/rcml/AttributeValidator.java

```diff
@@ -0,0 +1,548 @@
+package org.restcomm.connect.interpreter.rcml;
+
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.Logger;
+
+public interface AttributeValidator {
+
+	public boolean RecordValidator (Tag tag);
+	public boolean DialValidator (Tag tag);
+	public boolean EmailValidator (Tag tag);
+	public boolean PauseValidator (Tag tag);
+	public boolean SmsAndFaxValidator (Tag tag);
+	public boolean GatherValidator (Tag tag);
+	public boolean PlayAndSayValidator (Tag tag);
+	
+	
+}
+
+ class Validator implements AttributeValidator{
+	   Map <String, String> MyValidators = new TreeMap<String, String>();  
+	   private static Logger logger = Logger.getLogger(Parser.class);
+	   
+	  public Validator() {

```

Comment:
again we have a single validator impl class, with different specific methods to validate each verb. 

I would expect something like  a single interface, basic validators for length,regex,type, and then a Composite pattern  alllowing to combine basic validators, and impl class specialized in verb validations...
`
interface AttValidator {
public ValidationResult validate(Tag t);
}

class CompositeValidator implements AttValidator{
List<AttValidator> validators;

public ValidationResult validate(Tag t) {
ValidationResult result = new ValidationResult();
for (Validator v : validators) {
result.update(v.validate());
}
}

class LengthValidator implements AttValidator {
int maxSize;
int minSize;
String attName;
public ValidationResult validate(Tag t) {
....
}
}
Class DialValidator extends CompositeValidator {
public DialValidator() {
validators.add(new LngthValidator(...));
validators.add(new RegexValidator(..);
}
}
`